loopd/swapclient_server.go

```diff
@@ -579,15 +579,42 @@ func (s *swapClientServer) ListSwaps(ctx context.Context,
 		if !filterSwap(&swp, req.ListSwapFilter) {
 			continue
 		}
-
 		rpcSwap, err := s.marshallSwap(ctx, &swp)
 		if err != nil {
 			return nil, err
 		}
 		rpcSwaps = append(rpcSwaps, rpcSwap)
 		idx++
 	}
-	return &looprpc.ListSwapsResponse{Swaps: rpcSwaps}, nil
+
+	filteredSwapCount := len(rpcSwaps)
```

Comment:
Ideally the pagination could be interleaved with the above filtering loop.

Reply:
Ok, I will look into this. It does seem ideal if I could do it that way.